### PR TITLE
OSPD13 upgrade README for 5.4.1 U8

### PR DESCRIPTION
**Pull request by jenniferchew**
New documentation for the process to upgrade OSPD 13 from 5.4.1 U5 to 5.4.1 U8 based on https://docs.google.com/document/d/1AW_VrXXSuXdZcWP1Ko01ds2oNu4pU4DMtizG4CGjk0o/edit

### CODE REVIEW

**Comment by jenniferchew** [comment]
@sunny-verma Please review the documentation for OSPD 13 5.4.1 U8

File: Documentation/Upgrade/README.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+===================================
+Minor Upgrade to Release 5.4.1 U8+
+===================================
+
+.. contents::
+   :local:
+   :depth: 3
+
+
+Use this documentation when upgrading between minor releases. The process applies to upgrades from Release 5.4.1 U5 to Release 5.4.1 U8+. During this process, the Nuage components and Red Hat can be updated at the same time.
```
Comment (krissterckx) [inline Documentation/Upgrade/README.rst]: U5 or U5+  (Question to Andre)

File: Documentation/Upgrade/README.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+It is assumed the operator is familiar with OpenStack upgrades, VSP installation, the distribution-specific installation and upgrade practices, and the specific requirements for operations in a production environment.
+
+
+Upgrade Paths
+-------------
+
+Upgrading to the Release 5.4.1 U8+ is supported only on OSPD-based deployments. In this release, you can upgrade only from Queens 5.4.1 U5 to Queens 5.4.1 U8.
```
Comment (krissterckx) [inline Documentation/Upgrade/README.rst]: This first sentence puzzles me . Also outside OSPD we support upgrade.

File: Documentation/Upgrade/README.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+
+Basic Configuration
+---------------------
+
+The basic configuration includes:
+   
+   * An existing installed VSD and VSC upgraded to latest VSP release
+   * A single OpenStack Controller node
+   * One or more Compute nodes (hypervisors) running the VRS with the latest VSP release
```
Comment (krissterckx) [inline Documentation/Upgrade/README.rst]: or no-VRS (SRIOV) nodes) , or AVRS nodes

File: Documentation/Upgrade/README.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    * Nuage_base_repo:
+        
+        - It contains Selinux-policy-nuage, nuage-bgp, Python-openvswitch-nuage, nuage-openstack-neutronclient, and nuage-puppet-modules.
+        - Enable this repository on the on OpenStack Controller nodes and on the Compute VRS or AVRS nodes.
+    
+    * Nuage_vrs_repo:
+        
+        - It contains nuage-metadata and nuage-openvswitch.
+        - Enable this repository on the OpenStack Controller node and the Nova Compute VRS nodes.
+
```
Comment (linahe93) [inline Documentation/Upgrade/README.rst]: We don't need to mention it is nova compute, or we make it also mentioned for Nuage_base_repo.

File: Documentation/Upgrade/README.rst [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+Before the Upgrade
+--------------------
+
+1. If you are upgrading an AVRS node, migrate all your VMs on the node to an AVRS Compute node that is not being upgraded. Perform the steps in the "Live Migrate a Virtual Machine" section in https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/13/html/director_installation_and_usage/migrating-virtual-machines-between-compute-nodes-osp#live-migrate-a-vm-osp
+
+2. Create the following repositories:
+    
+    * Nuage_base_repo:
+        
+        - It contains Selinux-policy-nuage, nuage-bgp, Python-openvswitch-nuage, nuage-openstack-neutronclient, and nuage-puppet-modules.
```
Comment (linahe93) [inline Documentation/Upgrade/README.rst]: an extra on here

File: Documentation/Upgrade/README.rst (hunk 1 above)
Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: I will check with Andre and update when I get an answer.

File: Documentation/Upgrade/README.rst (hunk 2 above)
Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: done - removed the first sentence

File: Documentation/Upgrade/README.rst (hunk 3 above)
Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: done

File: Documentation/Upgrade/README.rst (hunk 4 above)
Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: done - added to nuage_base_repo

File: Documentation/Upgrade/README.rst (hunk 5 above)
Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: done

**Comment by jenniferchew** [comment]
@krissterckx @linahe93 Thanks for your comments. I added them to the PR. 

@krissterckx Can you or @psairam review and merge this PR? Then @psairam and others can update this README for upgrade with their changes. Thanks.

File: Documentation/Upgrade/README.rst [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    
+    b. Download the tar files for the upgrade from https://github.com/nuagenetworks/nuage-ospdirector/releases
+    
+
+2. Make sure your all of the templates and environment files are updated with the environment values for your deployment.
+
+    Get the environment values from the /home/stack/nuage-ospdirector-bk directory and update all the templates and environment files for the deployment, such as neutron-nuage/nova-nuage/compute-avrs and templates/compute-avrs-role.yaml.
+
+
+3. To update the Overcloud image, follow these instructions: https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/13/html/keeping_red_hat_openstack_platform_updated/assembly-updating_the_overcloud
```
Comment (psairam) [inline Documentation/Upgrade/README.rst]: here we have steps to update overcloud deployments, updating overcloud image comes in step 4

File: Documentation/Upgrade/README.rst [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+
+Basic Configuration
+---------------------
+
+The basic configuration includes:
+   
+   * An existing installed VSD and VSC upgraded to latest VSP release
+   * A single OpenStack Controller node
+   * One or more Compute nodes (hypervisors) running the VRS, SR-IOV, or AVRS nodes with the latest VSP release
```
[the remark this answers is not in the document]
Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: done

File: Documentation/Upgrade/README.rst [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+--------------------
+
+1. If you are upgrading an AVRS node, migrate all your VMs on the node to an AVRS Compute node that is not being upgraded. Perform the steps in the "Live Migrate a Virtual Machine" section in https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/13/html/director_installation_and_usage/migrating-virtual-machines-between-compute-nodes-osp#live-migrate-a-vm-osp
+
+2. Create the following repositories:
+    
+    * Nuage_base_repo:
+        
+        - It contains Selinux-policy-nuage, nuage-bgp, Python-openvswitch-nuage, nuage-openstack-neutronclient, and nuage-puppet-modules.
+        - Enable this repository on the OpenStack Controller nodes, the Compute VRS or AVRS nodes, and the Nova Compute AVRS nodes.
```
[the remark this answers is not in the document]
Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: done

File: Documentation/Upgrade/README.rst [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+    
+    * Nuage_vrs_repo:
+        
+        - It contains nuage-metadata and nuage-openvswitch.
```
[the remark this answers is not in the document]
Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: done

File: Documentation/Upgrade/README.rst [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+        - Enable this repository on the OpenStack Controller node and the Nova Compute VRS nodes.
+        
+    
+    * Nuage_avrs_repo:
+        
+        - It contains nuage-metadata, nuage-openvswitch, nuage-selinux-avrs, and 6WIND packages.
```
[the remark this answers is not in the document]
Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: done

File: Documentation/Upgrade/README.rst (hunk 1 above)
Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: done

File: Documentation/Upgrade/README.rst (hunk 6 above)
Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: done

**Comment by jenniferchew** [comment]
@psairam I updated the PR with your comments.

File: Documentation/Upgrade/README.rst [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+    a. Back up the templates and environment files from /home/stack/nuage-ospdirector to /home/stack/nuage-ospdirector-bk.
+       
+       ::
+       
+           $ mv /home/stack/nuage-ospdirector /home/stack/nuage-ospdirector-bk
+       
+    
+    
+    b. Download the tar files for the upgrade from https://github.com/nuagenetworks/nuage-ospdirector/releases
+
```
Comment (psairam) [inline Documentation/Upgrade/README.rst]: Rewording point b
b. There are 2 ways to get the files for upgrade
i) Download the tar files for the upgrade from https://github.com/nuagenetworks/nuage-ospdirector/releases
(or)
ii) git clone https://github.com/nuagenetworks/nuage-ospdirector.git -b <release-tag>

Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: done

File: Documentation/Upgrade/README.rst [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+Use this documentation when upgrading between minor releases. The process applies to upgrades from Release 5.4.1 U5 to Release 5.4.1 U8+. During this process, the Nuage components can be updated at the same time.
+
+It is assumed the operator is familiar with OpenStack upgrades, VSP installation, the distribution-specific installation and upgrade practices, and the specific requirements for operations in a production environment.
+
+
+Upgrade Paths
+-------------
+
+In this release, you can upgrade only from Queens 5.4.1 U5 to Queens 5.4.1 U8.
```
Comment (krissterckx) [inline Documentation/Upgrade/README.rst]: That is what we agreed to QA, but i believe we claim U6 to U8 also.  So this statement is too limited to my view.

Comment (krissterckx) [reply Documentation/Upgrade/README.rst]: See line 46 also where we say "or later"

File: Documentation/Upgrade/README.rst [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+
+3. Get the latest Nuage docker images from the Red Hat Partner Registry by following these instructions: https://github.com/nuagenetworks/nuage-ospdirector/blob/OSPD13/Documentation/5.4.1/README.rst#phase-8-nuage-docker-containers
+
+Upgrade Workflow
+------------------
+
+1. Back up the configuration files for your deployment.
+    
+    In the following examples, the templates and environment files are in the /home/stack/nuage-ospdirector directory.
```
Comment (sunny-verma) [inline Documentation/Upgrade/README.rst]: Reword it: "In the following examples, the templates and environment files are in the /home/stack/nuage-ospdirector directory".

to:
" In the following example, all the templates and environment files for your deployment are in the /home/stack/nuage-ospdirector directory. We need to get new Nuage U8 release nuage-ospdirector/nuage-tripleoheat-templates so we ask user to back up before we replace existing one with new U8 codebase.

File: Documentation/Upgrade/README.rst [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+       ::
+       
+           $ mv /home/stack/nuage-ospdirector /home/stack/nuage-ospdirector-bk
+       
+    
+    
+    b. Get the tar files for the upgrade one of these ways:
+    
+       * Download them from https://github.com/nuagenetworks/nuage-ospdirector/releases
+       * Use ``git clone https://github.com/nuagenetworks/nuage-ospdirector.git -b``
```
Comment (sunny-verma) [inline Documentation/Upgrade/README.rst]: Please change: Use ``git clone https://github.com/nuagenetworks/nuage-ospdirector.git -b``
to:

 ``git clone https://github.com/nuagenetworks/nuage-ospdirector.git -b <release>``
for example:   ``git clone https://github.com/nuagenetworks/nuage-ospdirector.git -b osp-13.541U8.1``

File: Documentation/Upgrade/README.rst (hunk 12 above)
Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: updated line 27

File: Documentation/Upgrade/README.rst (hunk 13 above)
Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: done

File: Documentation/Upgrade/README.rst (hunk 14 above)
Comment (jenniferchew) [reply Documentation/Upgrade/README.rst]: done

**Comment by jenniferchew** [comment]
@psairam @sunny-verma Latest updates have been added.

**Pull request merged**